tasks/gen/mixins/migration_with_columns.cr

```diff
@@ -0,0 +1,70 @@
+module Gen::Mixins::MigrationWithColumns
+  SUPPORTED_TYPES = {"Bool", "Float64", "Int16", "Int32", "Int64", "String", "Time", "UUID"}
+
+  private def migrate_contents : String
+    String.build do |string|
+      string << "# Learn about migrations at: https://luckyframework.org/guides/database/migrations"
+      string << "\n"
+      string << "create table_for(#{resource_name}) do\n"
+      string << "  primary_key id : Int64\n"
+      string << "  add_timestamps\n"
+      columns.each do |column|
+        string << "  add #{column.name} : #{column.type}\n"
+      end
+      string << "end"
+    end
+  end
+
+  private def rollback_contents : String
+    "drop table_for(#{resource_name})"
+  end
+
+  private def columns : Array(Lucky::GeneratedColumn)
+    column_definitions.map do |column_definition|
+      column_name, column_type = parse_definition(column_definition)
+      Lucky::GeneratedColumn.new(name: column_name, type: column_type)
+    end
+  end
+
+  private def column_definitions
+    if column_arguments?
+      ARGV.skip(1)
+    else
+      [] of String
+    end
+  end
+
+  private def column_arguments? : Bool
+    !!ARGV[1]?
+  end
+
+  private def columns_are_valid?(required : Bool = false) : Bool
+    (!required || column_definitions.any?) && column_definitions.all? do |column_definition|
```

Comment:
Rather than passing in `required` true, you could change this to remove `column_definitions.any?` and instead do `column_definitions.all?`. That will return true if there are no columns so could be used in the model generator.

Then in the resource generator there could add the check there: `columns_are_valid? && column_definitions.any?`

That way `columns_are_valid?` does just one thing and the behavior is a bit clearer. Thoughts?

Reply:
You're right, that is a lot easier on the eyes.